src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java

```diff
@@ -54,15 +55,18 @@ private String getJudicialUserFullName(@NonNull String personalCode, IdamTokens
     }
 
     private static String splitInitials(JudicialUser judicialUser) {
-        String initials = judicialUser.getInitials();
-
-        if (isNotBlank(initials)) {
-            StringBuilder result = new StringBuilder();
-            for (int i = 0; i < initials.length(); i++) {
-                result.append(initials.charAt(i));
-                result.append(" ");
+        String fullName = judicialUser.getFullName();
+        String surname = judicialUser.getSurname();
+        String title = judicialUser.getTitle();
+
+        if (isNotBlank(fullName) && isNotBlank(surname) && isNotBlank(title)) {
+            String givenName = fullName.substring(title.length(), fullName.length() - surname.length()-1).trim();
+            String givenNameInitials = givenName.substring(0,1);
+            int extraNameInitialIndex = givenName.indexOf(" ") + 1;
+            if (extraNameInitialIndex != 0) {
+                givenNameInitials += " " + givenName.charAt(extraNameInitialIndex);
             }
-            return result.toString().trim();
+            return givenNameInitials.toUpperCase().trim();
```

Comment:
would it not have been simpler to use the initials field or is that not reliable?